mmv1/third_party/terraform/services/compute/data_source_google_compute_region_instance_group_manager.go

```diff
@@ -0,0 +1,63 @@
+package compute
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleComputeRegionInstanceGroupManager() *schema.Resource {
+
+	dsSchema := tpgresource.DatasourceSchemaFromResourceSchema(ResourceComputeRegionInstanceGroupManager().Schema)
+	tpgresource.AddOptionalFieldsToSchema(dsSchema, "name", "self_link", "project", "region")
+
+	return &schema.Resource{
+		Read:   dataSourceComputeRegionInstanceGroupManagerRead,
+		Schema: dsSchema,
+	}
+}
+
+func dataSourceComputeRegionInstanceGroupManagerRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	if selfLink, ok := d.GetOk("self_link"); ok {
```

Comment:
```suggestion
	if selfLink, ok := d.Get("self_link").(string); ok {
```

Could we do the type assertion here instead of using `GetOk`? Worried this might panic if `GetOk` returns another type for some reason.